### PR TITLE
Converted webpack-compilation-hash javascript file to typescript

### PR DESCRIPTION
**Pull request by chrisq21**
## Description

Converts `gatsby/src/redux/reducers/webpack-compilation-hash.js` to Typescript as `gatsby/src/redux/reducers/webpack-compilation-hash.ts`.

Added an interface called `ISetWebpackCompilationHashAction` in `gatsby/src/redux/types.ts`.

Updated `gatsby/src/redux/reducers/index.js` to pull in `webpackCompilationHashReducer` with `import` instead of `require`

## Related Issues

Related to the Typescript migration efforts #21995

### CODE REVIEW

File: packages/gatsby/src/redux/reducers/webpack-compilation-hash.ts [hunk 1]
```diff
@@ -0,0 +1,13 @@
+import { IGatsbyState, ISetWebpackCompilationHashAction } from "../types"
+
+export const webpackCompilationHashReducer = (
+  state: IGatsbyState["webpackCompilationHash"] = ``,
+  action: ISetWebpackCompilationHashAction
```
Comment (blainekasten) [inline packages/gatsby/src/redux/reducers/webpack-compilation-hash.ts:5]: this should be `ActionsUnion` as every reducer gets called for dispatches of any type.

**Comment by blainekasten** [comment]
Also you need to merge in master as we have conflicts

**Comment by chrisq21** [comment]
@blainekasten Sounds good! I just merged in master, resolved the conflicts, and added `ISetWebpackCompilationHashAction` to `ActionsUnion` 👍

File: docs/blog/2020-04-27-building-Serverless-site-faster-with-Gatsby/index.md [hunk 2]
```diff
@@ -14,6 +14,7 @@ Since 2015 Serverless has been providing a platform to unify all the tools softw
 Any new front-end technologies had to mesh well with the existing site’s architecture. “The Serverless site has a blog and a documentation section, both of which pull Markdown files from GitHub repositories,” says Narayan. “We needed a way to pull those Markdown files while still maintaining a very fluent experience for the content contributors.”
 
 ## Ready to (re)launch with Gatsby 🚀
+
```
Comment (chrisq21) [inline docs/blog/2020-04-27-building-Serverless-site-faster-with-Gatsby/index.md:17]: Prettier was throwing errors for the missing the line breaks that are added here. @blainekasten Let me know if you'd like me to remove them.

**Pull request merged**